packages/web/src/services/apiClient.ts

```diff
@@ -18,6 +21,10 @@ if (!process.env.API_ORIGIN) {
 
 export const client = createTRPCClient<AppRouter>({
   url: process.env.API_ORIGIN,
+  fetch: typeof window === 'undefined' ? (f as any) : window.fetch.bind(window),
+  headers: {
+    Origin: process.env.CORS_ORIGIN,
```

Comment:
I'm not sure if the api need to throw a 500 error and check for origin. A browser has a built-in mechanisms to check CORS.

So I guess the public api implementation should set `Access-Control-Allow-*` headers and do not throw a 500 error

Reply:
Probably this is not related, too, though fine-tuning of error handling on public-api end might be good idea.